### PR TITLE
Hw 9

### PR DESCRIPTION
**Pull request by KovalAlexey**

### CODE REVIEW

**Review by Victoria-Moiseienko — COMMENTED** [review]
Good job!

File: src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java [hunk 1]
```diff
@@ -162,12 +166,17 @@ private void setUserRoles(User user, Connection connection)
...
         String userLogin = resultSet.getString("user_login");
         String userPassword = resultSet.getString("password");
-        User user = new User(userName, userLogin, userPassword, getUserRole(userId, connection));
+        byte[] userSalt = resultSet.getBytes("user_salt");
+        User user =
+                new User(userName,
+                        userLogin,
+                        userPassword,
+                        getUserRole(userId, connection),
+                        userSalt);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java]: let's use setters here

File: src/main/java/com/internet/shop/model/User.java [hunk 2]
```diff
@@ -15,11 +16,12 @@ public User(String name, String login, String password) {
...
     }
 
-    public User(String name, String login, String password, Set<Role> roles) {
+    public User(String name, String login, String password, Set<Role> roles, byte[] salt) {
         this.name = name;
         this.login = login;
         this.password = password;
         this.roles = roles;
+        this.salt = salt;
     }
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/model/User.java]:
```suggestion
    public User(String name, String login, String password, Set<Role> roles, byte[] salt) {
        this(name, login, password)
        this.roles = roles;
        this.salt = salt;
    }
```

File: src/main/resources/init_db.sql [hunk 3]
```diff
@@ -20,6 +20,7 @@ CREATE TABLE `internet-shop`.`users`
     `user_login` VARCHAR(256) NOT NULL,
     `password`   VARCHAR(256) NOT NULL,
     `is_deleted` TINYINT      NULL DEFAULT 0,
+    `user_salt` BLOB NOT NULL,
```
Comment (dmytro-hryhoruk) [inline src/main/resources/init_db.sql]:
```suggestion
    `salt` BLOB NOT NULL,
```
the table name is user. We don't need to specify this in column names too

File: src/main/java/com/internet/shop/model/User.java (hunk 2 above)
Comment (KovalAlexey) [reply src/main/java/com/internet/shop/model/User.java]: done

File: src/main/resources/init_db.sql (hunk 3 above)
Comment (KovalAlexey) [reply src/main/resources/init_db.sql]: done

File: src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java (hunk 1 above)
Comment (KovalAlexey) [reply src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java]: done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**